nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteStreamCommand.java

```diff
@@ -287,6 +311,32 @@ protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String
         .build();
     }
 
+    @OnScheduled
+    public void setupExecutor(final ProcessContext context) {
+        executor = Executors.newFixedThreadPool(context.getMaxConcurrentTasks() * 2, new ThreadFactory() {
+            private final ThreadFactory defaultFactory = Executors.defaultThreadFactory();
+
+            @Override
+            public Thread newThread(final Runnable r) {
+                final Thread t = defaultFactory.newThread(r);
+                t.setName("ExecuteStreamCommand " + getIdentifier() + " Task");
+                return t;
+            }
+        });
+    }
+
+    @OnUnscheduled
```

Comment:
@rkarthik29 

OnUnscheduled runs before all the executing threads are complete, I think it would be safer to use OnStopped to shutdown the executor